src/main/java/uk/ac/ebi/eva/contigalias/dus/PassiveAnonymousFTPClient.java

```diff
@@ -86,4 +93,25 @@ public void disconnect() throws IOException {
         }
     }
 
+    @Retryable(value = Exception.class, maxAttempts = 5, backoff = @Backoff(delay = 100, maxDelay = 500))
+    public boolean downloadFTPFile(String ftpFilePath, Path downloadFilePath, long ftpFileSize) throws IOException {
+        super.setFileType(FTP.BINARY_FILE_TYPE);
+        super.setFileTransferMode(FTP.BINARY_FILE_TYPE);
+        Files.deleteIfExists(downloadFilePath);
+
+        boolean success = super.retrieveFile(ftpFilePath, new FileOutputStream(downloadFilePath.toFile()));
+
+        if (success && Files.exists(downloadFilePath) && Files.isReadable(downloadFilePath)) {
+            long downloadFileSize = Files.size(downloadFilePath);
+            if (ftpFileSize == downloadFileSize) {
```

Comment:
I don't know how reliable this method is... Is there a way to compare checksums instead?

Reply:
In my experience File size is very reliable as long as the file is not tiny. 
I don't think we would have access to a file digest anyway.